### PR TITLE
[CI] image build logic enhancement

### PR DESCRIPTION
**Pull request by SamYuan1990**
- use latest GHA to manage container image build instead of make script.
- update logic between base image and image itself
- add arm64 and s390 image support

### CODE REVIEW

**Comment by SamYuan1990** [comment]
> @SamYuan1990 The changes mostly look good to me. Is it possible to have the base image rebuild only when the relevant files are changed (Dockerfile.base and requirements.txt)? https://how.wtf/run-workflow-step-or-job-based-on-file-changes-github-actions.html

I will try, there two items.
1. if the base image updates, we need to build from base image.
2. if there no changes on base image, we need to build model server image any way.

but, I am curious that it seems before this PR. if we update (for example, requirement.txt) and I didn't find a line in code limit the model server image build after base image?

**Comment by SamYuan1990** [comment]
`ERROR: failed to solve: quay.io/sustainable_computing_io/kepler_model_server_base:v0.7: pulling from host quay.io failed with status code [manifests v0.7]: 502 Bad Gateway` @sunya-ch could you please help rerun the test case? https://github.com/sustainable-computing-io/kepler-model-server/actions/runs/7086391503/job/19284563239?pr=199

**Comment by SamYuan1990** [comment]
@sunya-ch , could you please help review this PR?

**Review by sunya-ch — APPROVED** [review]
/lgtm Thank you.

File: .github/workflows/build-push.yml [hunk 1]
```diff
@@ -2,42 +2,84 @@ name: BuildPushDeployImage
...
+    branches: [ main ]
 
 jobs:
-  build:
+
+  changes:
     runs-on: ubuntu-latest
+    outputs:
+      src: ${{ steps.changes.outputs.src }}
+      infra: ${{ steps.changes.outputs.infra }}
```
Comment (sunya-ch) [inline .github/workflows/build-push.yml:13]: I don't see infra map below. 
I think we need to remove it.

**Review by sunya-ch — CHANGES_REQUESTED** [review]
I don't think we need `infra` in outputs. Other parts are good to go.

File: .github/workflows/build-push.yml (hunk 1 above)
Comment (SamYuan1990) [reply .github/workflows/build-push.yml]: updated.

**Review by sunya-ch — APPROVED** [review]
Thank you!

**Pull request merged**